File: backend/schemas/debate.py
"""
Debate Pydantic schemas for Multi-AI Debate Agent API.
"""

from typing import List, Optional, Dict
from datetime import datetime
from pydantic import BaseModel, Field


class DebateCreate(BaseModel):
    """Schema for creating a new debate."""
    topic: str = Field(..., min_length=1, max_length=500, description="辩论主题")
    max_rounds: int = Field(default=3, ge=1, le=10, description="最大辩论轮次")
    models: Dict[str, str] = Field(
        default_factory=lambda: {"pro": "mimo", "con": "deepseek", "judge": "mimo"},
        description="模型配置"
    )


class MessageResponse(BaseModel):
    """Schema for message response."""
    id: str
    debate_id: str
    round_number: int
    role: str
    content: str
    model_used: str
    confidence: float
    created_at: Optional[str] = None

    class Config:
        from_attributes = True


class VerdictResponse(BaseModel):
    """Schema for verdict response."""
    recommendation: str
    winner: str
    confidence: float
    action_plan: List[str]


class DebateResponse(BaseModel):
    """Schema for debate response."""
    id: str
    topic: str
    status: str
    created_at: Optional[str] = None
    completed_at: Optional[str] = None
    verdict: Optional[VerdictResponse] = None
    action_plan: Optional[List[str]] = None
    messages: Optional[List[MessageResponse]] = None

    class Config:
        from_attributes = True


class DebateListResponse(BaseModel):
    """Schema for debate list response."""
    debates: List[DebateResponse]
    total: int
    page: int
    page_size: int

File: debate-agent/frontend/src/api/debate.ts
/**
 * Debate API functions.
 */

import { api } from "./client";

export interface DebateCreate {
  topic: string;
  max_rounds?: number;
  models?: Record<string, string>;
}

export interface MessageResponse {
  id: string;
  debate_id: string;
  round_number: number;
  role: string;
  content: string;
  model_used: string;
  confidence: number;
  created_at?: string;
}

export interface VerdictResponse {
  recommendation: string;
  winner: string;
  confidence: number;
  action_plan: string[];
}

export interface DebateResponse {
  id: string;
  topic: string;
  status: string;
  created_at?: string;
  completed_at?: string;
  verdict?: VerdictResponse;
  action_plan?: string[];
  messages?: MessageResponse[];
}

export interface DebateListResponse {
  debates: DebateResponse[];
  total: number;
  page: number;
  page_size: number;
}

export const debateApi = {
  create: (data: DebateCreate) => api.post<DebateResponse>("/debates", data),
  list: (page = 1, pageSize = 10) =>
    api.get<DebateListResponse>(`/debates?page=${page}&page_size=${pageSize}`),
  get: (id: string) => api.get<DebateResponse>(`/debates/${id}`),
  start: (id: string) => api.post<DebateResponse>(`/debates/${id}/start`),
};
